src/client/service-components/EditableInput.js

```diff
@@ -0,0 +1,67 @@
+import React, {useState, useContext} from 'react';
+import {UserSettingsContext} from '../context-store/user-settings-context.js';
+import './EditableInput.css';
+
+export const EditableInput = (props) => {
+    const {theme} = useContext(UserSettingsContext);
+    const [isEdit, setIsEdit] = useState(false);
+    const [isInputEmpty, setIsInputEmpty] = useState(false);
```

Comment:
nit: I would name this flag `hasErrors` in case you have validation or some other errors for inputs

Reply:
Done. Commit 136989fa35278d64e7bbb3c84f2a8166394ae4a9